### ISSUE TITLE
Generic form containers for non-input elements and for custom Angular components.

### ISSUE THREAD

**Issue by whizkidwwe1217** [vmware/clarity#2886]
## Summary
**Angular Form Input Controls**

1. Allow non-input elements such as h1, h2, span, etc. to be laid out nicely with labels in the standard clarity form design.
2. Containers or wrappers for custom components in forms.

## Use case

**1. Non-input element/static content**
There are some fields in a form that don't need an input from the user and the purpose of those fields is just to display read-only or system generated texts. These static contents needs to be laid out properly in the form just like how the labels are aligned and the existing supported input controls are being laid out in the form.

**2. Custom components**
Right now, only those input controls that have supported input directives like clrInput are supproted by Clarity. Without a directive, the control will not be displayed in the form. I should be able to display a custom component. For example, a rich text editor component.

## Examples

_Are there any examples in other projects that demonstrate this feature or enhancement?_
**Non-input components**
clr-input-container does not display an Angular component or an HTML element inside it if it doesn't have an input directive.
Example:
![image](https://user-images.githubusercontent.com/1977391/49065910-675e1e80-f25a-11e8-8084-53e851269a06.png)

```html
<clr-input-container>
    <label>Custom Component:</label>
    <span>INV-0002-200395</span>
</clr-input-container>
```
or
```html
<clr-input-container>
    <label>Custom Component:</label>
    <my-custom-component></my-custom-component>
</clr-input-container>
```
Tried without clr-input-container but labels and components are not laid out and formatted properly.
![image](https://user-images.githubusercontent.com/1977391/49065025-b6568480-f257-11e8-8a8f-1ffe87c3ba0e.png)
**Custom components**
![image](https://user-images.githubusercontent.com/1977391/49065283-7c39b280-f258-11e8-87c5-60603f8f65c9.png)

## Workarounds

_Is there a workaround for this feature at the moment, even if it is less than ideal?_
I'm currently using the Clarity forms for Angular. Maybe a combination of angular & Clarity's css for forms

**Comment by Jinnie** [comment]
I believe this duplicates #2864
I won't be directly closing it, please review and comment if the approach discussed there will suit your needs.

**Comment by gnomeontherun** [comment]
Thank you for the well written request, this is useful for helping understand the use cases.

As a workaround, you could use the HTML and build the same DOM structure, but there could still be edge cases and some CSS to apply to make things look exactly as you wish. This is definitely something we need to look into handling, but since this one is more detailed I'm going to mark the other as a duplicate and close it in favor of this one.

**Comment by paddyfink** [comment]
Hello,
I haven't tried it out but does that mean that I cannot use for example kendo ui component inside `clr-input-container`?

**Comment by gnomeontherun** [comment]
@paddyfink No you cannot, because they don't wire up the internals like we do. However, you could use the HTML/CSS version (the bottom part of the documentation for inputs for example) and replace the input with a third party component. There might be some styling issues to address for it to layout properly though.

**Comment by paddyfink** [comment]
Ok Thanks @gnomeontherun. I will give it a try.

**Comment by rbirkgit** [comment]
This one is important. Even having a simple container object that doesn't support validation etc would be very useful.

**Comment by dennitsa** [comment]
I was planning on opening a new request, but perhaps I should just add our use case here.
It would be nice to be able to add extra content inside an input element(or select, checkbox and so on).
E.g. we have inputs that require the user to type in minutes and we want to have the text "minutes" right after the input field. Using helper text is not an option in this case, as it is already filled with info explaining the purpose of the input field. E.g. We would like to be able to have something like this:
![custom elements inside inputs](https://user-images.githubusercontent.com/17265957/62361984-bca6aa00-b524-11e9-9d50-2d0a33647242.png)

Also using elements such as singpost after the input is something we have in our existing UI and currently the only option is to move the signpost content inside the helper field, but our UX has concerns with this approach.

**Comment by dorthrithil** [comment]
@dennitsa Input groups are probably what you are looking for. There's already a separate issue for this: https://github.com/vmware/clarity/issues/1478

**Comment by dennitsa** [comment]
> @dennitsa Input groups are probably what you are looking for. There's already a separate issue for this: #1478

As far as I can see the other issue hasn't been considered in Clarity 2.0+ either. Anyway, whether it is a custom component or a standard input with extra custom content next to it, I think the use case is quite similar.

**Comment by rbirkgit** [comment]
We have a lot of custom components (and also use many signposts), and this is one of the major hurdles for us upgrading to the new forms.

**Comment by whizkidwwe1217** [comment]
I was able to make custom form controls and was able to make them work as if they are part of the pre-built Clarity forms. I created custom form controls and custom input containers. They respond very nicely with Clarity form's layout services, validations indicators, etc. The workaround is to copy all the required services and utility classes since Clarity doesn't expose these to the public. I just changed the filenames but kept the component/directive selectors unchanged. This allows me to trick Angular as if they are Clarity's. The only drawback that I can currently think of is that I have to keep this in-synced with the original code every time Clarity updates the library. Also, I'm not sure if there are adverse effects of doing this type of workaround.

![image](https://user-images.githubusercontent.com/1977391/71827421-2aaa5b00-30db-11ea-9c24-18ca71ec22fc.png)


_Sample custom search component_
![image](https://user-images.githubusercontent.com/1977391/71827095-727cb280-30da-11ea-8746-762284c654dd.png)

![image](https://user-images.githubusercontent.com/1977391/71826828-df437d00-30d9-11ea-88d7-02563ae48338.png)
